0004/README.md

```diff
@@ -0,0 +1,514 @@
+
+This is for a policy that lists explicit expectations for provenance values,
+instead of expectations for what provenance fields are invariant (regardless of
+their value).
+
+Inputs: 
+
+-  Package name and version.
+-  Optional: verified package provenance.
+-  Preferences. Examples include, but are not limited to:
+    -  Names of ‘sensitive' provenance properties that are not expected
+        to change such as builder ID or source repository,
+    -  Minimum SLSA level for an acceptable artifact, 
+    -  Maximum SLSA level depending on the builder. 
+
+-  Preferences may differ for different evaluator contexts: developer,
+    registry, or end user, and for a private or public registry.
+
+If no policy exists for this package, generate one with: 
+
+-  Package name 
+-  Version pattern that matches this package version and onwards.
+-  Expected provenance properties (SLSA level, signer, builder, source repo,
+    etc.).
+
+If a policy already exists for this package:
+
+-  Flag missing or unverifiable provenance.
+-  Flag an unexpected change in a sensitive provenance property (SLSA level,
+    source repo, etc.).
+
+The following will require approval:
+
+-  If a ‘good' package version does not satisfy expectations, add new
+    expectations, qualified with a version pattern that matches this package
+    version and onwards.
+
+## Main use cases
+
+### Use case 1. Publishing a new package version
+
+####  Existing process
+
+-  A developer runs "npm publish" to submit a new version of their npm
+    package to a registry.
+
+#### New behavior
+
+-  The package registry does a policy check for the package provenance,
+    initially non-enforcing.
+-  A private package registry may impose different expectations than a
+    public one. 
+
+#### New process
+
+-  A developer may also do a policy check in preparation for package upload.
+
+#### What packages will be covered by policy
+
+-  The policy applies only to the package that is being published (the
+    concrete versions of dependencies will be determined when a package is
+    downloaded for installation).
+
+####  Existing process details
+
+-  Besides package-specific code, an npm package contains a metadata file
+    (package.json) with the name and version of the package itself, and
+    optionally the names and allowed version ranges of direct dependencies.
+-  See "[Npm background](#heading=h.napkqzk41aim)" for more.
+
+### Use case 2. Installing a package locally
+
+The Appendix describes detailed scenarios to install packages and transitive
+dependencies. Despite differences in purpose, many are identical from a policy
+evaluation point of view:
+
+-  Installing all package dependencies of a project (Appendix
+    [A.1](#heading=h.vubk9ms4dtxh)).
+-  Adding a new package dependency to a project (Appendix
+    [A.2](#heading=h.vz2h8oar6gfd)).
+-  Installing an npm application locally (Appendix
+    [A.2](#heading=h.vz2h8oar6gfd)).
```

Comment:
Are these all Google Doc heading links?

Reply:
These were not updated by the Gdoc-to-Markdown conversion. I'll update these links too.